### PR TITLE
DOC, MAINT: Add issue template for Documentation issues

### PR DESCRIPTION
**Pull request by melissawm**
#### Reference issue
N/A

#### What does this implement/fix?
Adds a "Documentation" issue template.

#### Additional information
N/A

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/Documentation.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+body:
+- type: textarea
+  attributes: 
+    label: "Issue with current documentation:"
+    description: >
+      Please make sure to leave a reference to the document/code you're
+      referring to. You can also check the development version of the
+      documentation and see if this issue has already been addressed at
+      https://https://scipy.github.io/devdocs/.
```
Comment (mdhaber) [inline .github/ISSUE_TEMPLATE/Documentation.yml:14]: What do you think of emphasizing the part about checking the development version? We do get a fair number of reports about things that have already been fixed.

> Please check the development version of the documentation at https://https://scipy.github.io/devdocs/ to see if this issue has already been addressed. If not, please link to the development version of the document/code you're referring to.

File: .github/ISSUE_TEMPLATE/Documentation.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+- type: textarea
+  attributes:
+    label: "Idea or request for content:"
+    description: >
+      Please describe as clearly as possible what topics you think are missing
+      from the current documentation.
```
Comment (mdhaber) [inline .github/ISSUE_TEMPLATE/Documentation.yml:21]: Maybe invite suggestions for phrasing? Not that PR authors would be bound to it, but I think that might help us understand what the user is looking for.

**Review by mdhaber — COMMENTED** [review]
Thanks for this @melissawm. Do you know if we can preview what the form looks like before merging?

File: .github/ISSUE_TEMPLATE/Documentation.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Documentation
+description: Report an issue related to the SciPy documentation.
+title: "DOC: "
```
Comment (tupui) [inline .github/ISSUE_TEMPLATE/Documentation.yml:3]: I would also add something like [your title]. We do get empty PRs/issues

Comment (melissawm) [reply .github/ISSUE_TEMPLATE/Documentation.yml:3]: On NumPy we do `DOC: <Please write a comprehensive title after the 'DOC: ' prefix>`. We do still get empty titles though :sweat:

File: .github/ISSUE_TEMPLATE/Documentation.yml (hunk 2 above)
Comment (melissawm) [reply .github/ISSUE_TEMPLATE/Documentation.yml]: How about this?

File: .github/ISSUE_TEMPLATE/Documentation.yml (hunk 1 above)
Comment (melissawm) [reply .github/ISSUE_TEMPLATE/Documentation.yml]: Let me know if this is too much.

**Review by tupui — APPROVED** [review]
LGTM, Matt do you want other changes?

**Comment by mdhaber** [comment]
Besides the Oxford comma in "suggested fix, better phrasing, or topics", no. But that's a matter of personal preference, so I'll let it go : )

Thanks @melissawm @tupui!

**Pull request merged**